streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -508,13 +508,16 @@ public void shouldEnforceRebalanceAfterNextScheduledProbingRebalanceTime() throw
             10 * 1000,
             "Thread never started.");
```

Comment:
Thanks @ableegoldman . Do we still need the wait on L506-L509?

Reply:
Ah, I see. I was concerned that we could skip over the STARTING state, but the mock consumer won't let us. This LGTM, then.